### PR TITLE
improved support of cftime

### PR DESCRIPTION
**Pull request by TonioF**
A fix to further improve the usability of datasets with time given in a cftime format.
Checklist:

* [x] Add unit tests and/or doctests in docstrings
* ~~[ ] Add docstrings and API docs for any new/modified user-facing classes and functions~~
* ~~[ ] New/modified features documented in `docs/source/*`~~
* ~~[ ] Changes documented in `docs/CHANGES.md`~~
* [ ] AppVeyor and Travis CI passes
* [x] Test coverage remains or increases (target 100%)
* ~~[ ] Associated issues closed after merge~~

### CODE REVIEW

File: xcube/core/timecoord.py [hunk 1]
```diff
@@ -118,17 +118,14 @@ def get_time_range_from_data(dataset: xr.Dataset, maybe_consider_metadata: bool=
     time_diff = time.diff(dim=time.dims[0]).values
     time_res = time_diff[0]
     time_regular = all([time_res - diff == np.timedelta64(0) for diff in time_diff[1:]])
+    is_cf_time = type(time.values[0]).__module__.startswith('cftime')
```
Comment (forman) [inline xcube/core/timecoord.py:121]: 1. `time.values` eagerly loads all time data. 
2. it is not save to use `__module__` names.

```suggestion
    is_cf_time = instanceof(time[0].values.item(), cftime.datetime)
```

File: xcube/core/timecoord.py [hunk 2]
```diff
@@ -118,17 +118,14 @@ def get_time_range_from_data(dataset: xr.Dataset, maybe_consider_metadata: bool=
...
+    data_start = pd.to_datetime(time.values[0].isoformat()) if is_cf_time else time.values[0]
+    data_end = pd.to_datetime(time.values[-1].isoformat()) if is_cf_time else time.values[-1]
```
Comment (forman) [inline xcube/core/timecoord.py:123]: Every single `time.values`  call eagerly loads all time data. I explained the implications two weeks or so ago.

```suggestion
    data_start = pd.to_datetime(time[0].values.item().isoformat()) if is_cf_time else time[0].values.item()
    data_end = pd.to_datetime(time[-1].values.item().isoformat()) if is_cf_time else time[-1].values.item()
```

**Review by forman — APPROVED** [review]

**Review by forman — CHANGES_REQUESTED** [review]
Avoid eager loading of values.

**Comment by TonioF** [comment]
I have revised the method (and done a bit of formatting). Thanks for your input.

File: xcube/core/timecoord.py [hunk 3]
```diff
@@ -56,31 +57,32 @@ def add_time_coords(dataset: xr.Dataset, time_range: Tuple[float, float]) -> xr.
     time_var.encoding['calendar'] = DATETIME_CALENDAR
     if t1 != t2:
         time_var.attrs['bounds'] = 'time_bnds'
-        dataset = dataset.assign_coords(time_bnds=(['time', 'bnds'],
-                                                   from_time_in_days_since_1970([t1, t2]).reshape(1, 2)))
+        dataset = dataset.assign_coords(
+            time_bnds=(['time', 'bnds'], from_time_in_days_since_1970([t1, t2]).reshape(1, 2)))
```
Comment (forman) [inline xcube/core/timecoord.py:61]:
```suggestion
        dataset = dataset.assign_coords(
            time_bnds=(['time', 'bnds'], from_time_in_days_since_1970([t1, t2]).reshape(1, 2))
        )
```

PEP-8

File: xcube/core/timecoord.py [hunk 4]
```diff
@@ -189,15 +185,16 @@ def remove_time_part_from_isoformat(datetime_str: str) -> str:
...
+    timedelta = date_time - REF_DATETIME
     return timedelta.days + timedelta.seconds / SECONDS_PER_DAY + \
            timedelta.microseconds / MICROSECONDS_PER_DAY
 
 
 def from_time_in_days_since_1970(time_value: Union[float, Sequence[float]]) -> np.ndarray:
     if isinstance(time_value, int) or isinstance(time_value, float):
-        return pd.to_datetime(time_value, utc=True, unit='d', origin='unix').round(freq='ms').to_datetime64()
+        return pd.to_datetime(time_value, utc=True, unit='d', origin='unix').round(freq='ms').\
+            to_datetime64()
```
Comment (forman) [inline xcube/core/timecoord.py:197]:
```suggestion
        return pd.to_datetime(time_value, utc=True, unit='d', origin='unix').round(freq='ms') \
            .to_datetime64()
```

PEP-8

**Review by forman — CHANGES_REQUESTED** [review]
Just PEP-8

**Pull request merged**